pkg/gadgets/advise/seccomp/tracer/tracer.go

```diff
@@ -168,10 +179,14 @@ func (t *Tracer) DetachContainer(container *containercollection.Container) error
 	return nil
 }
 
-func (t *Tracer) Result() ([]byte, error) {
+func (t *Tracer) collectResult() ([]byte, error) {
 	out := make(map[string][]string)
 	for container, result := range t.containers {
 		out[container.Name] = result
 	}
 	return json.MarshalIndent(out, "", "  ")
 }
+
+func (t *Tracer) init(gadgetCtx gadgets.GadgetContext) error {
+	return nil
+}
```

Comment:
Actually, I forgot to remove all the `init`s that are empty (like this one) 😅 . I will do it.

I introduced it to make the parameters parsing and keep the `Run*` clean. In particular, because the one for the top gadget is quite long:

https://github.com/inspektor-gadget/inspektor-gadget/blob/0817ae862c7126af1428dc027d1dc1da0675b67a/pkg/gadgets/top/block-io/tracer/tracer.go#L376-L394

Maybe we could call it `parseParams()` and implement it only when the gadget has parameters. WDYT?

Reply:
Perhaps we could only have this for top gadgets. IMO it doesn't make sense in other gadgets as it's always few lines of code, so I feel it's cleaner to just put them directly on `Run*`. However this is just a personal preference, if you think it's better to have an independent function I'm fine with it.